File: service/java/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/BlockedListMessage.java
package org.whispersystems.signalservice.api.messages.multidevice;

import java.util.List;

public class BlockedListMessage {

  private final List<String> numbers;
  private final List<byte[]> groupIds;

  public BlockedListMessage(List<String> numbers, List<byte[]> groupIds) {
    this.numbers  = numbers;
    this.groupIds = groupIds;
  }

  public List<String> getNumbers() {
    return numbers;
  }

  public List<byte[]> getGroupIds() {
    return groupIds;
  }
}

File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/BlockedListMessage.kt
package org.whispersystems.signalservice.api.messages.multidevice

import org.whispersystems.signalservice.api.push.ServiceId

class BlockedListMessage(
  @JvmField val individuals: List<Individual>,
  @JvmField val groupIds: List<ByteArray>
) {
  data class Individual(
    val aci: ServiceId.ACI?,
    val e164: String?
  ) {
    init {
      check(aci != null || e164 != null)
    }
  }
}
